rpc/backend/tx_info_test.go

```diff
@@ -605,6 +608,179 @@ func (suite *BackendTestSuite) TestGetTransactionReceipt() {
+		{
+			"false- Receipts with a different chain ID not in the patched height. expected to return an invalid chain ID",
+			func() {
+
+				ctx := server.NewDefaultContext()
+
+				// overwrite client conext
+				suite.backend.clientCtx = suite.backend.clientCtx.WithChainID("canto_7700-1").WithHeight(notPatchedHeight)
+
+				idxer := indexer.NewKVIndexer(dbm.NewMemDB(), ctx.Logger, suite.backend.clientCtx)
+				suite.backend = NewBackend(ctx, ctx.Logger, suite.backend.clientCtx, true, idxer)
+				suite.backend.queryClient.QueryClient = mocks.NewEVMQueryClient(suite.T())
+				suite.backend.clientCtx.Client = mocks.NewClient(suite.T())
+				suite.backend.queryClient.FeeMarket = mocks.NewFeeMarketQueryClient(suite.T())
+				suite.backend.ctx = rpctypes.ContextWithHeight(notPatchedHeight)
+
+				var header metadata.MD
+				queryClient := suite.backend.queryClient.QueryClient.(*mocks.EVMQueryClient)
+				client := suite.backend.clientCtx.Client.(*mocks.Client)
+				RegisterParams(queryClient, &header, notPatchedHeight)
+				RegisterParamsWithoutHeader(queryClient, notPatchedHeight)
+				RegisterBlock(client, notPatchedHeight, txBz)
+				RegisterBlockResults(client, notPatchedHeight)
+			},
+			msgEthereumTx,
+			&types.Block{Header: types.Header{Height: notPatchedHeight}, Data: types.Data{Txs: []types.Tx{txBz}}},
+			&abci.ResponseFinalizeBlock{
+				TxResults: []*abci.ExecTxResult{
+					{
+						Code: 0,
+						Events: []abci.Event{
+							{Type: evmtypes.EventTypeEthereumTx, Attributes: []abci.EventAttribute{
+								{Key: "ethereumTxHash", Value: txHash.Hex()},
+								{Key: "txIndex", Value: "0"},
+								{Key: "amount", Value: "1000"},
+								{Key: "txGasUsed", Value: "21000"},
+								{Key: "txHash", Value: ""},
+								{Key: "recipient", Value: "0x775b87ef5D82ca211811C1a02CE0fE0CA3a455d7"},
+								{Key: "status", Value: "0x0"},
+							}},
+						},
+					},
+				},
+			},
+			map[string]interface{}(nil),
+			false,
+			false, // invalid chain id error
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			suite.SetupTest() // reset
+			tc.registerMock()
+
+			db := dbm.NewMemDB()
+			suite.backend.indexer = indexer.NewKVIndexer(db, log.NewNopLogger(), suite.backend.clientCtx)
+			err := suite.backend.indexer.IndexBlock(tc.block, tc.blockResult)
+			suite.Require().NoError(err)
+
+			txReceipt, err := suite.backend.GetTransactionReceipt(common.HexToHash(tc.tx.Hash))
+			if tc.expApiPass {
+				suite.Require().NoError(err)
+
+				if tc.expKeyMatchPass {
+
+					// check all expected receipt keys are exist.
+					for k, _ := range tc.expTxReceipt {
+						_, keyExist := txReceipt[k]
+						suite.Require().True(keyExist)
+					}
+
+				} else {
+
+					anyKeyNotExist := false
+					// check the receipt keys are exist.
+					for k, _ := range tc.expTxReceipt {
+						_, keyExist := txReceipt[k]
+						if !keyExist {
+							anyKeyNotExist = true
+						}
+
+					}
+					suite.Require().True(anyKeyNotExist)
+				}
```

Comment:
There is no testcase where `expApiPass` is true and `expKeyMatchPass` is false, so this is code block that is not currently being entered. I would suggest adding a testcase or removing the check logic.